### PR TITLE
MNT Remove unused JS and CSS

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1378

Kitchen-sink CI - https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/10657647066

### CODE REVIEW

File: client/src/legacy/LeftAndMain.Menu.js
Comment (ChloeSartorelli) [inline client/src/legacy/LeftAndMain.Menu.js:1]: There's still a reference that
> The (optional) second level is collapsible, hiding its children.

There's also still a lot of references like `li.children.opened`, `this.hasClass('collapsed')`, `<span class="toggle-children">` etc, a `toggle` function, `open` function, and `close` function...  that all look like they're related to this legacy feature.

File: client/src/components/Menu/Menu.scss
Comment (ChloeSartorelli) [inline client/src/components/Menu/Menu.scss:1]: There's still stuff like `.toggle-children` that look like they're related to this legacy feature.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: client/src/legacy/LeftAndMain.Menu.js
Comment (emteknetnz) [reply client/src/legacy/LeftAndMain.Menu.js:1]: Updated

File: client/src/components/Menu/Menu.scss
Comment (emteknetnz) [reply client/src/components/Menu/Menu.scss:1]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM, works as expected.

**Pull request merged**